contracts/EthereumForkArbiter.sol

```diff
@@ -0,0 +1,53 @@
+pragma solidity 0.4.15;
+
+import './AccessControl/AccessControlled.sol';
+import './AccessRoles.sol';
+import './Reclaimable.sol';
+
+
+contract EthereumForkArbiter is AccessControlled, AccessRoles, Reclaimable {
+
+    string public nextForkName;
+    string public nextForkUrl;
+    uint256 public lastSignedBlockNumber;
+    bytes32 public lastSignedBlockHash;
+    uint256 public lastSignedTimestamp;
+
+    event ForkAnnounced(
+        string name,
+        string url
+    );
+
+    event ForkSigned(
+        uint256 blockNumber,
+        bytes32 blockHash
+    );
+
+    function EthereumForkArbiter(IAccessPolicy accessPolicy)
+        AccessControlled(accessPolicy)
+        Reclaimable()
+    {
+    }
+
+    /// @notice Announce that a particular future Ethereum fork will be considered the valid one. Once the fork has happened, it will eventually be confirmed by signing a block on the fork. Notice that forks may happen unannounced.
+    function announceFork(string name, string url)
+        public
+        only(ROLE_FORK_ARBITER)
+    {
+        nextForkName = name;
+        nextForkUrl = url;
+        ForkAnnounced(nextForkName, nextForkUrl);
+    }
+
+    /// @notice Declare that the current fork (as identified by a blockhash) is the valid fork.
+    function signFork(uint256 number, bytes32 hash)
+        public
+        only(ROLE_FORK_ARBITER)
+    {
+        require(block.blockhash(number) == hash);
+        lastSignedBlockNumber = number;
+        lastSignedBlockHash = hash;
+        lastSignedTimestamp = block.timestamp;
+        ForkSigned(lastSignedBlockNumber, lastSignedBlockHash);
+    }
+}
```

Comment:
Conceptual problem: how this is intended to be used? Should we aggregate arbiter in each contract we care about forks? Like Neumark and LockedAccount?

Reply:
Just one globally should be enough, that's how I deploy it.

Reply:
Multiple would be a bad idea, they could only conflict.

Reply:
They don't need to be linked to the contracts. I did not want to put in a `kill all bad forks` mechanism for security reasons.

Only thing we need to do is put the contract address in the agreements in the paragraph that says how to identify the one-true-fork.

Reply:
Yes, there should be just one, what I meant is that you keep its address in any contract that has agreement attached. In that case you have the same mechanism for legal and smart contract - a link to fork arbiter... We may think on adding arbiter with mixin that adds url to legal agreement. Let's discuss tomorrow

Reply:
I like the idea of connecting contracts back to the ForkArbiter. It also seems to overlap with the Agreement contract. I want to combine the two, so let's get this PR in and finish it in Agreement one, ok?